### PR TITLE
fix(ui): bump dependencies

### PR DESCRIPTION
**Pull request by lesnitsky**
## Description

See title

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_ui_auth/example/pubspec.yaml [hunk 1]
```diff
@@ -21,13 +21,9 @@ environment:
...
 
 dependencies:
-  cloud_firestore: ^4.0.1
-  crypto: ^3.0.1
   cupertino_icons: ^1.0.2
-  firebase_auth: ^3.10.0
-  firebase_core: ^1.10.3
-  firebase_database: ^10.0.0
-  firebase_dynamic_links: ^5.0.1
+  firebase_auth: ^4.0.0
```
Comment (russellwheatley) [inline packages/firebase_ui_auth/example/pubspec.yaml:25]: auth is 4.0.1. https://github.com/firebase/flutterfire/blob/master/packages/firebase_auth/firebase_auth/pubspec.yaml#L7
```suggestion
  firebase_auth: ^4.0.1
```

**Review by rrousselGit — APPROVED** [review]

File: packages/firebase_ui_auth/pubspec.yaml [hunk 2]
```diff
@@ -9,14 +9,10 @@ environment:
...
 dependencies:
-  cloud_firestore: ^4.0.1
-  collection: ^1.15.0
-  crypto: ^3.0.1
   desktop_webview_auth: ^0.0.9
   email_validator: ^2.0.1
-  firebase_auth: ^3.10.0
-  firebase_core: ^1.10.2
-  firebase_database: ^10.0.0
+  firebase_auth: ^4.0.0
```
Comment (russellwheatley) [inline packages/firebase_ui_auth/pubspec.yaml:14]:
```suggestion
  firebase_auth: ^4.0.1
```

**Review by russellwheatley — CHANGES_REQUESTED** [review]

**Comment by russellwheatley** [comment]
Also curious why core is needed in auth but not firestore and database? Is it not transitive?

**Comment by russellwheatley** [comment]
Does this need bumping to 10.0.1? https://github.com/firebase/flutterfire/blob/50794a8cd41e20071bd82cdee15133a1849e9f4e/packages/flutterfire_ui/pubspec.yaml#L22

**Comment by russellwheatley** [comment]
and this to 4.0.1? https://github.com/firebase/flutterfire/blob/50794a8cd41e20071bd82cdee15133a1849e9f4e/packages/flutterfire_ui/pubspec.yaml#L22

**Review by russellwheatley — APPROVED** [review]

**Pull request merged**